src/Symfony/Component/Form/Tests/Extension/Core/Type/TextTypeTest.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\Tests\Extension\Core\Type;
+
+use Symfony\Component\Form\Test\TypeTestCase as TestCase;
+
+class TextTypeTest extends TestCase
+{
+    public function testSubmitNullReturnsNull()
+    {
+        $form = $this->factory->create('Symfony\Component\Form\Extension\Core\Type\TextType', 'name');
+
+        $form->submit(null);
+
+        $this->assertNull($form->getData());
+    }
+
+    public function testSubmitNullReturnsEmptyStringWithEmptyDataAsString()
+    {
+        $form = $this->factory->create('Symfony\Component\Form\Extension\Core\Type\TextType', 'name', array(
+            'empty_data' => '',
+        ));
+
+        $form->submit(null);
```

Comment:
with "new system" i was refering to the state of the form component after your PR.

i was specially talking about how to distinguish between a text field that is submitted with null vs submitted with empty string.

consider a simple form with text field:
```php
$form = $factory->create();
$form->add('name', TextType::class);
```
and now the different submits:
```php
// 1.
$form->submit(null);
assert $form->get('name')->getData() === null;

// 2.
$form->submit([]);
assert $form->get('name')->getData() === null;

// 3.
$form->submit([ 'name' => null ]);
assert $form->get('name')->getData() === null;

// 4.
$form->submit([ 'name' => '' ]);
assert $form->get('name')->getData() === '';

// 5.
$form->submit([ 'name' => 'Joe' ]);
assert $form->get('name')->getData() === 'Joe';
```

- `name` is truely optional here, but is also allowed to be the empty string (that is also different from "name not given")
- 1. can be viewed as a submission error, though one could normalize it to `[]`